libvast/src/system/tracker.cpp

```diff
@@ -118,10 +118,59 @@ behavior terminator(stateful_actor<terminator_state>* self, caf::error reason,
   };
 }
 
+void put_component(scheduled_actor* self, tracker_state& st,
```

Comment:
How about calling this function `register_component` for more semantics?

Reply:
This function is called from the message handlers for `put_atom`, hence the `put_` prefix to make it recognizable from the actor interface. The actor interface currently uses `put` and `get`. Would you like that to change to `register` and `get`?

Reply:
> Would you like that to change to `register` and `get`?

No, that doesn't strike me as a reasonable improvement. It would make the actor interface more complicated due to the absence of a natural duality of `put` and `get`. 

I consider `put` and `get` atoms primarily relevant to the actor interface, because these are commonly used for store-like interfaces. Internally, the implementation is not bound to these terms. Hence my suggestion to call this helper function `register_component` because "putting" a component is less meaningful than "registering" it.